docs/madr/decisions/009-tracing-policy.md

```diff
@@ -0,0 +1,237 @@
+[`overall_sampling`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-tracing-overall-sampling)
+via the `sampling` field.
+
+This MADR proposes instead the following:
+
+```yaml
+spec:
+ default:
+  sampling:
+   overall: <percentage>
+   client: <percentage>
+   random: <percentage>
+```
+
+with the justification that it's not clear to users
+that when they set the simple `sampling` field we have now,
+whether traces started via `x-client-trace-id` are limited.
+This also enables disabling `x-client-trace-id`.
+
+#### Tags
+
+Envoy can set tags in traces using values from:
+
+- Literal value
+- Request header
+- Environment variable
+- Metadata
+
+We propose allowing users to configure custom tags using only literal values or
+request headers. Using Kuma leaves environment variables and Envoy metadata
+opaque for users so the use cases for configuring them (at least directly)
+in Kuma are limited.
+
+Tags can be configured as follows:
+
+```yaml
+spec:
+ default:
+  tags:
+   - name:
+     # exactly one of the following keys must be set
+     literal: ...
+     header:
+      name:
+      default:
+```
+
+#### Method/path specific
+
+Users can set the `spec.targetRef` to be a `MeshGatewayRoute` or
+`MeshTrafficRoute`.
+
+Both paths and methods are matchable in `MeshGatewayRoute` and we can ensure
+that `MeshTrafficRoute` also supports this matching.
+
+## Examples
+
+All examples assume Kubernetes.
+
+### Simple
+
+This configures all `Dataplane` inbounds with `kuma.io/service: backend` to
+trace a maximum of 80% of traffic and adds the custom tag `team` with a value of
+`core` to spans. The backend is specified in a `TracingBackend` object.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Tracing
+metadata:
+ name: all
+ labels:
+  kuma.io/mesh: default
+spec:
+ targetRef:
+  kind: MeshService
+  name: backend
+ default:
+  backend:
+   ref:
+    kind: TracingBackend
+    name: jaeger
+  sampling:
+   overall: 80
+  tags:
+   - name: team
+     literal: core
+```
+
+### Inline backend configuration
+
+This specifies the backend directly:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Tracing
+metadata:
+ name: all
+ labels:
+  kuma.io/mesh: default
+spec:
+ targetRef:
+  kind: MeshService
+  name: backend
+ default:
+  backend:
+   zipkin:
+    url: http://jaeger-collector.mesh-observability:9411/api/v2/spans
+  sampling:
+   overall: 80
+```
+
+### Route-specific
+
+This configures any listeners matched by the `MeshGatewayRoute` `prod`
+to trace a maximum of 80% of traffic and adds the custom tag `env`
+with a value from `x-env` and a default value of `prod`.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Tracing
+metadata:
+ name: all
+ labels:
+  kuma.io/mesh: default
+spec:
+ targetRef:
+  kind: MeshGatewayRoute
+  name: prod
+ default:
+  backend:
+   ref:
```

Comment:
is that the same format as in MeshTrafficLog?

Reply:
I provocatively changed `reference` to `ref` but otherwise I think so. Though I just saw that `reference` was removed from the MADR so I'm not sure anymore.

Reply:
You can do a further PR to fix the log madr to use ref instead of reference